apps/extension/src/routes/page/page-layout.tsx

```diff
@@ -0,0 +1,40 @@
+import { Outlet } from 'react-router-dom';
+import { useStore } from '../../state';
+import { passwordSelector } from '../../state/password';
+import { useEffect } from 'react';
+import { PagePath } from './paths';
+import { usePageNav } from '../../utils/navigate';
+import { accountsSelector } from '../../state/accounts';
+
+const getLocationHash = (location: string, key: PagePath) =>
+  (location.replace('#', '') as PagePath) === key;
+
+export const PageLayout = () => {
```

Comment:
Why should this logic wrap every single page? This whole component does not feel like the right approach.

Reply:
Why it does not feel like the right approach?

Reply:
There we have logic, what checking is you login

Reply:
we need to wrap everything so that a user with an open page and popup can log in/out of the popup or page, and the page and popup states should change together! maybe you have another idea how to do it?

Reply:
User access to pages: 
1. ❌password ❌accounts =>  Welcome, Welcome-import, Welcome-set-password, welcome-generate, welcome-confirm-backup pages; 
2.  ❌password ✅accounts => Login, Restore password, Restore-password-set-password pages
3. ✅password ✅accounts => Index, welcome-success pages

Reply:
Gotcha. See `pageIndexLoader`. I think there is overlap here. Let's implement this logic as a loader.

Reply:
And what about data synchronization between popup and tab?

Reply:
Ah ok, I see what you mean. Ok, let's keep this as a hook then. More questions:

With hash router, we don't need this right?
```typescript
  useEffect(() => {
    // redirect to reset wallet after popup forgot password click
    if (getLocationHash(window.location.hash, PagePath.RESTORE_PASSWORD))
      navigate(PagePath.RESTORE_PASSWORD);
  }, []);
``` 

This line too right?

```typescript
if (getLocationHash(window.location.hash, PagePath.RESTORE_PASSWORD)) return;
```

Shouldn't it be like this?

```typescript
   if (!all.length) {
      navigate(PagePath.WELCOME);
    } else if (!hashedPassword) {
      navigate(PagePath.LOGIN);
    }
```

Because if someone has a password, we don't want to redirect them everytime to the index page.

Reply:
With hash router, we don't need this
```
 useEffect(() => {
    // redirect to reset wallet after popup forgot password click
    if (getLocationHash(window.location.hash, PagePath.RESTORE_PASSWORD))
      navigate(PagePath.RESTORE_PASSWORD);
  }, []);
```
and this
`if (getLocationHash(window.location.hash, PagePath.RESTORE_PASSWORD)) return;`

We should check all version
```
  if (!all.length) {
      navigate(PagePath.WELCOME);
    } else if (!hashedPassword) {
      navigate(PagePath.LOGIN);
    }else if(hashedPassword){
      navigate(PagePath.INDEX)}
```

Reply:
what differents between success and index pages?

Reply:
Because page-layout wraps all pages, doesn't that mean if they have a password, they'll be guaranteed to always be redirected to index? Like what if they are in the settings page or doing something else. You don't want to redirect those users.

Reply:
yes, but we still have a situation where the user is on the login page, and here we should be redirected to the index when the login was performed from another context

Reply:
That is an edge case I'd say. But what about the case I mentioned? That is far more common. How would we handle that if we are always redirect users to index?

Reply:
add this to useEffect with hashedPassword and isInit, add the value isInit, as an example, which will become true when we create the first account, because it's a bad idea to look at the length of the array of accounts (there will be many in the future)

Reply:
I just don't see a case where a user has two login screens open. One in the popup and one in the page. When a user is logging in, they are 99% going to be doing so through the popup.

Reply:
We need to think about such issues, because there will be users who want to break something! our tester, for example)

Reply:
or refuse tab mode

Reply:
This solution: https://github.com/penumbra-zone/web/pull/20#discussion_r1306034616, does not address the 99% case. Let's say a user on the popup clicks settings. They get directed to the page, but they have a password already and that means they get redirected to the index page. Essentially, this prevents users from getting to deeper pages.

Reply:
```
export const pageRoutes: RouteObject[] = [
  {
    element: <PageLayout />,
```

```

export const PageLayout = () => {
  const { hashedPassword } = useStore(passwordSelector);

  useEffect(() => {
    console.log(hashedPassword);
  }, [hashedPassword]);

  return <Outlet />;
};
```
no, it works only when the hashedPassword value changes

Reply:
In this case, let's drop this case:

```
if(hashedPassword) navigate(PagePath.INDEX)}
```

Adds complexity with almost no benefit.

Reply:
Is it possible to create a 3 layouts? for each case

Reply:
Ah, if we care about that specific use case for the login page, let's just put the useEffect there and not in the layout component for every page.

Reply:
I'm sorry, I've been on the project for a long time now, and I use tab and popup only for develop and test new feature at the same time! You're right, new users will use 1 of the contexts, not 2 together